### PR TITLE
Change getDirectoryDeviceId to getDeviceSerialNumber due to cros66 change

### PR DESCRIPTION
**Pull request by msnorman343**

### CODE REVIEW

**Comment by grahamgilbert** [comment]
We should have `getDirectoryDeviceId` as a "fact" as well. https://github.com/salopensource/sal/wiki/Checkin-and-checkin-modules

**Comment by msnorman343** [comment]
Great idea!  Let me take a look at this.

File: popup.js [hunk 1]
```diff
@@ -245,6 +245,7 @@ function sal4ReportFormat(report){
   new_report = {
       'serial': data.serial,
       'hostname': data.serial,
+      'device_id': data.google_device_identifier,
```
Comment (grahamgilbert) [inline popup.js]: I don't see where this is put into the new Sal 4 format - am I missing it somewhere? https://github.com/salopensource/sal/wiki/Checkin-and-checkin-modules

**Review by grahamgilbert — CHANGES_REQUESTED** [review]

**Comment by msnorman343** [comment]
Let me know if anything further is needed.

File: popup.js [hunk 2]
```diff
@@ -260,7 +260,10 @@ function sal4ReportFormat(report){
   out.Sal = {}
   out.Chrome.facts = {'checkin_module_version': data.sal_version}
   out.Sal.facts = {'checkin_module_version': data.sal_version}
-  out.Machine.facts = {'checkin_module_version': data.sal_version}
+  out.Machine.facts = {
+    'checkin_module_version': data.sal_version,
+    'Google_Device_ID': data.google_device_identifier
```
Comment (grahamgilbert) [inline popup.js]: This should be lower case

File: popup.js [hunk 3]
```diff
@@ -362,6 +365,46 @@ function sendData(){
...
+      console.log('Not a managed chrome device');
+      if (debug === true){
+        console.log(err);
+      }
+      if (debug === false) {
+        console.log('setting do not send to true due to no serial error and not being debug')
+        doNotSend = true;
+      }
+    }
+    callbackCount++;
```
Comment (grahamgilbert) [inline popup.js:404]: You’re increasing the callback count but not increasing the callback total

**Review by grahamgilbert — CHANGES_REQUESTED** [review]

**Review by grahamgilbert — APPROVED** [review]

**Pull request merged**